tsdb/engine/tsm1/compact.go

```diff
@@ -1088,7 +1088,11 @@ func (c *Compactor) RemoveTmpFiles(files []string) error {
 func (c *Compactor) RemoveTmpFilesOnErr(files []string, originalErrs ...error) error {
 	removeErr := c.RemoveTmpFiles(files)
 	if removeErr == nil {
-		return errors.Join(originalErrs...)
+		if len(originalErrs) == 1 {
```

Comment:
Is this required? Wouldn't `errors.Join(originalErrs...)` capture a single error?

Reply:
It does, but this is a quick hack to allow older code looking for a specific error by casting instead of using `errors.Is` to continue to function.